docs/enhancements/cluster-deployment-condition-refactor.md

```diff
@@ -0,0 +1,95 @@
+# Hive ClusterDeployment Conditions Refactor
+
+## Summary
+
+Reduce the number of conditions on ClusterDeployment and provide callers better options to monitor progress.
+
+## Motivation
+
+ClusterDeployment now carries over 20 conditions by default, from a variety of controllers. This is difficult for users to parse and there is not presently one condition that can be monitored for overall progress of an install.
+
+## Goals
+
+  1. Reduce the number of conditions on ClusterDeployment.
+  1. Provide callers high level conditions they can monitor.
+
+
+## Background
+
+Current 22 conditions:
+
+- Unreachable
+- ClusterImageSetNotFound
+- ControlPlaneCertificateNotFound
+- Hibernating
+- RelocationFailed
+- SyncSetFailed
+- AWSPrivateLinkFailed
+- AWSPrivateLinkReady
+- ActiveAPIURLOverride
+- AuthenticationFailure
+- ClusterInstallCompleted
+- ClusterInstallFailed
+- ClusterInstallRequirementsMet
+- ClusterInstallStopped
+- DNSNotReady
+- DeprovisionLaunchError
+- IngressCertificateNotFound
+- InstallImagesNotResolved
+- InstallLaunchError
+- InstallerImageResolutionFailed
+- ProvisionFailed
+- ProvisionStopped
+
+## Proposal
+
+Focusing only on the 12 conditions controlled by the clusterdeployment controller:
+
+- ClusterImageSetNotFound - Move to RequirementsMet or possibly ClusterInstallRequirementsMet.
+- AuthenticationFailure - Move to RequirementsMet or ClusterInstallRequirementsMet.
```

Comment:
Aren't we merging `ClusterInstallRequirementsMet` with `RequirementsMet`?

Reply:
Good catch, updating.